src/instance-config.ts

```diff
@@ -0,0 +1,496 @@
+   * @property {object[]} 0 The metadata of the instance config.
+   * @property {string} 0.name The unique identifier for the instance config.
+   * @property {string} 0.displayName The name of the instance config as it
+   *     appears in UIs.
+   * @property {google.spanner.admin.instance.v1.IReplicaInfo[]} 0.replicas The replicas used by
+   *     this instance config.
+   * @property {string[]} 0.leaderOptions The possible leader options for this instance config.
+   */
+  /**
+   * @callback GetInstanceConfigCallback
+   * @param {?Error} err Request error, if any.
+   * @param {object} instanceConfig The metadata of the instance config.
+   * @param {string} instanceConfig.name The unique identifier for the instance
+   *     config.
+   * @param {string} instanceConfig.displayName The name of the instance config
+   *     as it appears in UIs.
+   * @param {google.spanner.admin.instance.v1.IReplicaInfo[]} instanceConfig.replicas The replicas used by
+   *     this instance config.
+   * @param {string[]} 0.leaderOptions The possible leader options for this instance config.
+   */
+  /**
+   * Get a specific instance config.
+   *
+   * Wrapper around {@link v1.InstanceAdminClient#getInstanceConfig}.
+   *
+   * @see {@link v1.InstanceAdminClient#getInstanceConfig}
+   * @see [GetInstanceConfig API Documentation](https://cloud.google.com/spanner/docs/reference/rpc/google.spanner.admin.instance.v1#google.spanner.admin.instance.v1.InstanceAdmin.GetInstanceConfig)
+   *
+   * @param {GetInstanceConfigCallback} [callback] Callback function.
+   * @returns {Promise<GetInstanceConfigResponse>}
+   *
+   * @example
+   * ```
+   * const {Spanner} = require('@google-cloud/spanner');
+   * const spanner = new Spanner();
+   *
+   * const instanceConfig = spanner.instanceConfig('custom-my-instance-config');
+   *
+   * instanceConfig.get(function(err, instanceConfig) {
+   *   // `instanceConfig` is an instance configuration descriptor.
+   * });
+   *
+   * //-
+   * // If the callback is omitted, we'll return a Promise.
+   * //-
+   * instanceConfig.get().then(function(data) {
+   *   const instanceConfig = data[0];
+   * });
+   * ```
+   */
+  get(options?: GetInstanceConfigOptions): Promise<GetInstanceConfigResponse>;
+  get(callback: GetInstanceConfigCallback): void;
+  get(
+    options: GetInstanceConfigOptions,
+    callback: GetInstanceConfigCallback
+  ): void;
+  get(
+    optionsOrCallback?: GetInstanceConfigOptions | GetInstanceConfigCallback,
+    cb?: GetInstanceConfigCallback
+  ): void | Promise<GetInstanceConfigResponse> {
+    const callback =
+      typeof optionsOrCallback === 'function' ? optionsOrCallback : cb!;
+    const options =
+      typeof optionsOrCallback === 'object'
+        ? optionsOrCallback
+        : ({} as GetInstanceConfigOptions);
+    this.parent.getInstanceConfig(this.id, options, callback);
+  }
+
+  /**
+   * Update the metadata for this instance config. Note that this method follows
```

Comment:
Maybe add something like: This function can only be called for custom instance configs.

Reply:
Done.